src/cli/lint/internalLints/upgrade/patches/9.0.1/patchPnpmDockerImages.ts

```diff
@@ -0,0 +1,83 @@
+import { inspect } from 'util';
+
+import fg from 'fast-glob';
+import { readFile, writeFile } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { log } from '../../../../../../utils/logging';
+
+const DOCKER_IMAGE_CONFIG_REGEX =
+  /^(RUN )(pnpm config set store-dir \/root\/.pnpm-store)/gm;
+const DOCKER_IMAGE_FETCH_REGEX =
```

Comment:
Tries to look for the patched line:

```
--mount=type=bind,source=package.json,target=package.json
```

 inbetween

```Dockerfile
RUN --mount=type=bind,source=.npmrc,target=.npmrc
```

and 

```bash
pnpm fetch
```

Reply:
This is challenging to review but the tests look good!